contracts/truefi2/TrueCreditLine.sol

```diff
@@ -1,35 +1,116 @@
+     */
+    mapping(address => uint256) public previousCumulatedRewards;
+
+    // track claimable rewards for accounts
+    mapping(address => uint256) public claimableRewards;
+
+    uint256 public totalInterestRewards;
+    uint256 public totalClaimedRewards;
 
     /**
      * @dev Create Credit Line
      * @param _borrower Borrower address
-     * @param _pool Pool for which the credit line is attached to
-     * @param _amount Amount of tokens to be minted to the pool
+     * @param _pool Pool to which the credit line is attached to
+     * @param _principalDebt Initial amount of debt taken by borrower
      */
     constructor(
         address _borrower,
-        address _pool,
-        uint256 _amount
+        ITrueFiPool2 _pool,
+        uint256 _principalDebt
     ) public {
         ERC20.__ERC20_initialize("TrueFi Credit Line", "tfCL");
 
         borrower = _borrower;
         pool = _pool;
-        _mint(_pool, _amount);
+
+        principalDebt = _principalDebt;
+        _mint(address(_pool), _principalDebt);
+    }
+
+    /**
+     * @dev Update state of rewards
+     */
+    modifier update(address account) {
+        // calculate total rewards
+        uint256 newTotalInterestRewards = token().balanceOf(address(this)).add(totalClaimedRewards);
+        // calculate new reward
+        uint256 totalNewRewards = newTotalInterestRewards.sub(totalInterestRewards);
+        // update interest rewards
+        totalInterestRewards = newTotalInterestRewards;
+        cumulativeTotalRewards = cumulativeTotalRewards.add(totalNewRewards);
+        // update claimable reward for sender
+        claimableRewards[account] = claimableRewards[account].add(
+            balanceOf(account).mul(cumulativeTotalRewards.sub(previousCumulatedRewards[account])).div(totalSupply())
+        );
+        // update previous cumulative for sender
+        previousCumulatedRewards[account] = cumulativeTotalRewards;
+        _;
+    }
+
+    /**
+     * @dev View to estimate the claimable reward for an account
+     * @return claimable rewards for account
+     */
+    function claimable(address account) external view returns (uint256) {
+        // calculate total rewards (including not updated)
+        uint256 newTotalInterestRewards = token().balanceOf(address(this)).add(totalClaimedRewards);
+        // calculate new reward
+        uint256 totalNewRewards = newTotalInterestRewards.sub(totalInterestRewards);
+        // return claimable reward for this account
+        // prettier-ignore
+        return claimableRewards[account].add(
+            balanceOf(account).mul(cumulativeTotalRewards.add(totalNewRewards).sub(previousCumulatedRewards[account])).div(totalSupply())
+        );
+    }
+
+    function transfer(address recipient, uint256 amount) public virtual override update(msg.sender) update(recipient) returns (bool) {
+        super.transfer(recipient, amount);
```

Comment:
Instead of overriding both `transfer` and `transferFrom` you can just override `_ transfer`. Btw you forgot to return true